prov/sockets/src/sock_av.c

```diff
@@ -188,7 +192,19 @@ static int sock_check_table_in(struct sock_av *_av, struct sockaddr_in *addr,
 							context, i, FI_ENOMEM);
 						continue;
 					}
+					_av->idx_arr[_av->table_hdr->stored] = _av->table_hdr->stored;
+					_av->table_hdr = new_addr;
+					_av->table_hdr->size = new_count;
+					_av->table = (struct sock_av_addr *)
+						((char *)_av->table_hdr +
+						idx_arr_sz +
+						sizeof(struct sock_av_table_hdr));
```

Comment:
#1651 had a bug and it passed the shared-av test by chance. This patch handles the calculation properly. I'll cleanup the code and push an update.